### PR TITLE
feat: Component for listing the portal

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

Fixes https://github.com/chatwoot/product/issues/499

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

| **Storybook view**                                                                                                                                     | **Props and Events**                                                                                                                                      |
| ------------------------------------------------------------------------------------------------------------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------ |
| Screenshot 2022-08-02 at 1 56 27 PM | Screenshot 2022-07-29 at 11 07 08 AM |

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/dashboard/i18n/locale/en/helpCenter.json [hunk 1]
```diff
@@ -37,6 +37,39 @@
...
+              "SUB_TEXT": "Portal sub text"
+            }
+          },
+          "AVAILABLE_LOCALES": {
+            "TITLE": "Available locales",
+            "TABLE": {
+              "NAME": "Locale name",
+              "CODE": "Locale code",
+              "ARTICLE_COUNT": "No. of articles",
+              "CATEGORES": "No. of categores",
```
Comment (muhsin-k) [inline app/javascript/dashboard/i18n/locale/en/helpCenter.json:66]:
```suggestion
              "CATEGORIES": "No. of categories",
```

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue [hunk 2]
```diff
@@ -0,0 +1,415 @@
+<template>
+  <div>
+    <div v-for="portal in portals" :key="portal.id" class="portal">
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue:3]: Create the Portal Item as a separate component.

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue [hunk 3]
```diff
@@ -0,0 +1,415 @@
...
+      <thumbnail :username="portal.name" variant="square" />
+      <div class="container">
+        <header>
+          <div>
+            <div class="title-status--wrap">
+              <h2 class="portal-title">{{ portal.name }}</h2>
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue:9]: Use existing typography classes

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue [hunk 4]
```diff
@@ -0,0 +1,415 @@
...
+              size="small"
+              icon="settings"
+              class="header-action-buttons"
+              color-scheme="secondary"
+              @click="openSettings"
+            />
+          </div>
+        </header>
+        <div class="portal-locales">
+          <h2 class="locale-title">
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue:65]: Typography classes here too

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue [hunk 5]
```diff
@@ -0,0 +1,415 @@
...
+                <div class="content-theme-wrap">
+                  <div
+                    class="theme-color"
+                    :style="{ background: portal.color }"
+                  />
+                  <span class="content">{{ portal.page_title }}</span>
+                </div>
+              </div>
+              <div class="configuration-item">
+                <span class="heading">{{
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue:125]: Can we use `label` for this type of title? There are other occurrences too

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue [hunk 6]
```diff
@@ -0,0 +1,415 @@
...
+        </div>
+        <div class="portal-locales">
+          <h2 class="locale-title">
+            {{
+              $t(
+                'HELP_CENTER.PORTAL.PORTAL_SETTINGS.LIST_ITEM.AVAILABLE_LOCALES.TITLE'
+              )
+            }}
+          </h2>
+          <table>
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue:143]: Can we move this table as a separate component.

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue [hunk 7]
```diff
@@ -0,0 +1,415 @@
...
+  background-color: var(--white);
+  padding: var(--space-normal);
+  border: 1px solid var(--color-border-dark);
+  border-radius: var(--border-radius-medium);
+  position: relative;
+  display: flex;
+  margin-bottom: var(--space-normal);
+
+  .container {
+    margin-left: var(--space-one);
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue:310]: Use space small

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue [hunk 8]
```diff
@@ -0,0 +1,415 @@
...
+    header {
+      display: flex;
+      align-items: flex-start;
+      justify-content: space-between;
+      margin-bottom: var(--space-normal);
+      .title-status--wrap {
+        display: flex;
+        align-items: center;
+        .status {
+          margin: 0 0 0 var(--space-one);
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue:322]: Use margin left and `space-small`

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue [hunk 9]
```diff
@@ -0,0 +1,415 @@
...
+      }
+      .content {
+        font-size: var(--font-size-small);
+      }
+      .content-theme-wrap {
+        display: flex;
+        align-items: center;
+      }
+      .theme-color {
+        width: var(--space-two);
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue:386]: Use size from Figma

**Review by nithindavid — DISMISSED** [review]

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue (hunk 2 above)
Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue:3]: It's already in a separate component.

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue (hunk 6 above)
Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue:143]: Will do it

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue (hunk 8 above)
Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue:322]: @nithindavid  The label component has some margins on the bottom and right that's why I added it. 

image

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue (hunk 6 above)
Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue]: Done https://github.com/chatwoot/chatwoot/pull/5146/commits/dfb0fee4025c93e61807f42e2afe3a6b802fc038

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue (hunk 2 above)
Comment (muhsin-k) [reply app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue:3]: @iamsivin Can't we iterate `portals` array in `AllPortal` component itself?

Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue:3]: In this PR we are not linking it with routes thats why i don't add it. Will add it.

Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalListItem.vue:3]: @muhsin-k I think we use it in `app/javascript/dashboard/routes/dashboard/helpcenter/pages/portals/ListAllPortals.vue` this file or here it self.

**Review by muhsin-k — APPROVED** [review]

**Pull request merged**